api/src/main/java/org/ccci/idm/user/User.java

```diff
@@ -411,12 +412,15 @@ public void setCruProxyAddresses(Collection<String> cruProxyAddresses) {
         this.cruProxyAddresses = cruProxyAddresses;
     }
 
-    public Collection<String> getCruPasswordHistory() {
-        return cruPasswordHistory;
+    public Set<String> getCruPasswordHistory() {
+        return ImmutableSet.copyOf(cruPasswordHistory);
     }
 
-    public void setCruPasswordHistory(final Collection<String> cruPasswordHistory) {
-        this.cruPasswordHistory = cruPasswordHistory;
+    public void setCruPasswordHistory(@Nullable final Collection<String> history) {
+        cruPasswordHistory.clear();
+        if (history != null) {
+            cruPasswordHistory.addAll(history);
```

Comment:
here we take all the entries out of whatever collection is provided (e.g. `ArrayList$SubList`) and add it to our own internal collection that will be serializable